### ISSUE TITLE
Weekly Paper 7. 리사이징, 정규화, 데이터 증강, 전이 학습

### ISSUE THREAD

**Issue by finneKIM** [finneKIM/WeeklyPaper#8]
<html>
<body>
<h1>이미지를 모델에 입력하기 전에 리사이징(Resizing)과 정규화(Normalization)를 하는 이유는 무엇인가요?</h1>
<h2>✅ 1. 리사이징 (Resizing)</h2>
<blockquote>
<p>이미지의 크기를 동일하게 맞추는 과정</p>
</blockquote>
<h3>📌 왜 해야 할까?</h3>
<ol>
<li><strong>모든 이미지 크기가 제각각이라서</strong>
<ul>
<li>예: 고양이 사진은 500x400 픽셀, 강아지 사진은 1280x720 픽셀 등</li>
<li>👉 모델에 입력하려면 크기가 같아야 계산이 가능함</li>
</ul>
</li>
<li><strong>모델은 고정된 입력 크기를 요구함</strong>
<ul>
<li>대부분의 CNN(합성곱 신경망) 모델은 <code>[Height, Width, Channels]</code> 형태로 <strong>고정된 크기</strong>의 이미지를 받아들임</li>
<li>예: <code>224x224x3</code> (ImageNet 기반 모델의 표준 입력)</li>
</ul>
</li>
<li><strong>연산량 통제 및 메모리 효율성 확보</strong>
<ul>
<li>큰 이미지를 그대로 넣으면 <strong>메모리 사용량 급증</strong></li>
<li>예: 1024x1024 → 224x224로 축소하여 <strong>모델 속도 향상</strong></li>
</ul>
</li>
</ol>
<hr>
<h2>✅ 2. 정규화 (Normalization)</h2>
<blockquote>
<p>픽셀 값의 범위를 일정하게 조정하는 과정</p>
</blockquote>
<h3>📌 왜 해야 할까?</h3>
<h3>✔️ (1) 픽셀 값의 범위가 너무 크기 때문</h3>
<ul>
<li>일반적으로 이미지의 픽셀 값은 <strong>0~255</strong>
<ul>
<li>0 = 검정, 255 = 흰색</li>
</ul>
</li>
<li>→ 그대로 사용하면 <strong>값의 범위가 너무 크고, 모델 학습이 불안정함</strong></li>
</ul>
<h3>✔️ (2) 수치가 클수록 <strong>기울기 폭발(gradient exploding)</strong> 등의 문제가 생김</h3>
<ul>
<li>신경망은 가중치를 업데이트할 때 **미분값(기울기)**을 사용</li>
<li>값이 너무 크면 가중치가 너무 많이 바뀌거나, 너무 적게 바뀜</li>
</ul>
<h3>✔️ (3) 빠른 학습, 안정된 수렴</h3>
<ul>
<li>정규화를 통해 값의 범위를 <strong>[0, 1]</strong> 또는 **[-1, 1]**로 조정하면
<ul>
<li>모델이 더 빠르고 안정적으로 학습함</li>
</ul>
</li>
</ul>
<hr>
<h2>✅ 정규화 방법 예시</h2>
<pre><code class="language-python">
transforms.ToTensor()
# 0~255 정수 → 0~1 실수로 나눔

transforms.Normalize(mean=[0.5, 0.5, 0.5],
                     std=[0.5, 0.5, 0.5])
# 0~1 → -1 ~ 1 로 조정

</code></pre>
<hr>
<h2>✅ 예시 시각화</h2>

원본 이미지 (500x400) | 리사이즈 후 (224x224) | 정규화 후
-- | -- | --
픽셀 범위: 0~255 | 크기 동일: 224x224 | 픽셀 범위: -1~1
다양한 해상도 | 모델에 맞게 고정됨 | 계산 안정성 확보

</body>
</html>

**Comment by finneKIM** [comment]
<html>
<body>
<h1>데이터 증강(Data Augmentation)이란 무엇이며, 이미지 데이터에서 주로 사용하는 증강 기법에는 어떤 것들이 있나요?</h1>
<h2>✅ 1. 데이터 증강이란?</h2>
<blockquote>
<p>학습 데이터를 인위적으로 다양하게 만드는 기술</p>
</blockquote>
<h3>📌 왜 필요할까?</h3>
<ol>
<li><strong>데이터가 부족할 때 효과적</strong>
<ul>
<li>딥러닝 모델은 많은 데이터를 요구하지만, 직접 수집하는 건 시간/비용 부담이 큼</li>
<li>기존 데이터를 변형해서 <strong>더 많은 데이터가 있는 것처럼 학습 가능</strong></li>
</ul>
</li>
<li><strong>모델의 일반화 성능 향상</strong>
<ul>
<li>같은 이미지만 반복해서 학습하면 과적합(훈련 데이터에만 맞춰짐)이 발생할 수 있음</li>
<li>증강된 다양한 버전을 학습시키면 <strong>새로운 이미지에도 잘 반응</strong>하게 됨</li>
</ul>
</li>
<li><strong>현실 상황 반영</strong>
<ul>
<li>예: 고양이 사진이 항상 정중앙, 밝은 조명, 똑바로 선 상태로만 있지 않음</li>
<li>회전, 뒤집기, 밝기 변화 등을 적용해 <strong>현실 상황에서 모델이 잘 작동</strong>하게 만들 수 있음</li>
</ul>
</li>
</ol>
<hr>
<h2>✅ 2. 이미지에서 자주 사용하는 데이터 증강 기법</h2>

종류 | 설명 | 예시 효과
-- | -- | --
수평 뒤집기 (Horizontal Flip) | 좌우 반전 | 오른쪽을 보고 있던 고양이가 왼쪽을 보게 됨
회전 (Rotation) | 이미지를 일정 각도 돌림 | 15도 회전된 이미지 생성
자르기 (Random Crop) | 이미지의 일부분만 사용 | 배경 일부 잘림, 물체 위치 변경
확대/축소 (Zoom) | 크기를 조정함 | 물체가 커지거나 작아짐
밝기 조절 (Brightness) | 이미지의 밝기 증가/감소 | 조명이 강하거나 약한 환경을 모방
대조 조절 (Contrast) | 밝고 어두운 정도 차이 조정 | 선명도 변경
색상 조절 (Hue, Saturation) | 색조나 채도를 변경 | 조명에 따라 색깔이 바뀐 상황 반영
노이즈 추가 (Gaussian Noise) | 이미지에 작은 잡음 추가 | 촬영 시 생기는 흔들림이나 잡음 재현
Affine 변환 | 이동, 확대, 회전 등을 합친 변형 | 다양한 시점에서의 시각 정보 제공



</body>
</html>

**Comment by finneKIM** [comment]
<html>
<body>
<h1>Transfer Learning(전이 학습)이란 무엇이며, 이미지 분류 모델에서 어떻게 활용할 수 있나요?</h1>
<h2>✅ 1. 전이 학습(Transfer Learning)이란?</h2>
<blockquote>
<p>기존에 학습된 모델의 가중치를 새로운 작업에 일부 또는 전체 재사용하는 학습 방식</p>
</blockquote>
<hr>
<h3>📌 쉽게 설명하면?</h3>
<ul>
<li>대규모 데이터셋(예: ImageNet)으로 잘 훈련된 모델이 있다고 가정해보자.</li>
<li>이 모델은 **기본적인 시각 정보(선, 색, 모양, 텍스처 등)**을 잘 파악하도록 훈련되어 있음.</li>
<li>이제 이 모델을 <strong>다른 작업</strong>(예: 고양이 vs 강아지 분류)에 활용하면,
<ul>
<li>처음부터 다시 학습할 필요 없이 <strong>이미 배운 것을 활용해서 빠르게 학습 가능</strong>함.</li>
</ul>
</li>
</ul>
<hr>
<h2>✅ 2. 왜 전이 학습을 사용할까?</h2>

이유 | 설명
-- | --
데이터가 부족할 때 | 수천~수만 장의 이미지를 모으기 어려운 경우 효과적
학습 시간이 짧음 | 이미 많은 것을 배운 모델을 쓰므로, 학습 시간이 줄어듦
좋은 성능 | 처음부터 학습한 것보다 일반적으로 더 나은 성능을 냄


<hr>
<h2>✅ 4. 실전 예시 (PyTorch 기준)</h2>
<h3>💡 사전 학습된 모델 불러오기 (예: ResNet18)</h3>
<pre><code class="language-python">
from torchvision import models
import torch.nn as nn

# 1. 사전 학습된 모델 불러오기
model = models.resnet18(pretrained=True)  # ImageNet으로 학습됨

# 2. 기존 가중치 고정 (특징 추출 방식)
for param in model.parameters():
    param.requires_grad = False

# 3. 마지막 출력층만 바꿔줌 (예: 고양이/강아지 → 클래스 2개)
model.fc = nn.Linear(model.fc.in_features, 2)

</code></pre>
<blockquote>
<p>이 구조로 학습하면 이미지는 ResNet이 분석하고, 마지막 출력층만 내 문제에 맞게 새로 학습함</p>
</blockquote>
<hr>
<h2>✅ 5. 전이 학습의 시각적 구조</h2>
<pre><code>
[사전학습된 ResNet 모델]
 ⬇️ conv layers (엣지/모양 등 기본 특성 추출)
 ⬇️ global features (복잡한 패턴 이해)
 ⬇️ (기존 출력층 제거)
 ⬇️ [새 출력층: 내 문제에 맞게 수정]

</code></pre>

</body>
</html>
